x/ccv/provider/keeper/grpc_query.go

```diff
@@ -49,26 +50,57 @@ func (k Keeper) QueryConsumerChains(goCtx context.Context, req *types.QueryConsu
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	chains := []*types.Chain{}
-	for _, chainID := range k.GetAllRegisteredConsumerIds(ctx) {
-		c, err := k.GetConsumerChain(ctx, chainID)
+	lastCID, ok := k.GetConsumerId(ctx)
+	if !ok {
+		return &types.QueryConsumerChainsResponse{}, nil
+	}
+
+	consumerIDs := []string{}
+
+	// return the consumer chains in the "Launched" phase by default
+	phase := Launched
+
+	// update phase if filter flag is  set
+	if req.FilterByPhase {
+		phase = ConsumerPhase(byte(req.Phase))
+	}
+
+	switch {
+	case phase == Launched:
+		consumerIDs = append(consumerIDs, k.GetAllRegisteredConsumerIds(ctx)...)
```

Comment:
Why `append` here and below. Why not simply `consumerIds = k.GetAllRegisteredConsumerIds...`